File: src/main/java/com/tonitealive/server/auth/Roles.java
package com.tonitealive.server.auth;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.core.env.Environment;
import org.springframework.stereotype.Component;

@Component("roles")
public final class Roles {

    public final String USER;
    public final String ADMIN;
    public final String PROMOTER;

    @Autowired
    public Roles(Environment environment) {
        USER = environment.getProperty("stormpath.authorized.group.user");
        ADMIN = environment.getProperty("stormpath.authorized.group.admin");
        PROMOTER = environment.getProperty("stormpath.authorized.group.promoter");
    }

}

File: src/main/kotlin/com/tonitealive/server/auth/Roles.kt
package com.tonitealive.server.auth

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.core.env.Environment
import org.springframework.stereotype.Component

@Component("roles")
class Roles @Autowired constructor(val environment: Environment) {

    val USER: String
    val ADMIN: String
    val PROMOTER: String

    init {
        USER = environment.getProperty("stormpath.authorized.group.user")
        ADMIN = environment.getProperty("stormpath.authorized.group.admin")
        PROMOTER = environment.getProperty("stormpath.authorized.group.promoter")
    }

}

